### PR TITLE
Only pay attention to origin types in project lifecycle admission

### LINKED ISSUE #11094: LimitRange defaults flake
```
• Failure in Spec Teardown (AfterEach) [300.360 seconds]
[k8s.io] LimitRange
/data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:793
  should create a LimitRange with defaults and ensure pod has those defaults applied. [AfterEach]
  /data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/limit_range.go:102

  Sep 26 03:54:52.649: Couldn't delete ns: "e2e-tests-limitrange-c4yb1": namespace e2e-tests-limitrange-c4yb1 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed (&errors.errorString{s:"namespace e2e-tests-limitrange-c4yb1 was not deleted with limit: timed out waiting for the condition, namespace is empty but is not yet removed"})

  /data/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:338
```

as seen in:
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6466/consoleFull

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #11094:
* origin finalizer is now only added for creation of origin types
* namespace finalizer is moved to the front of the chain, and already does a lot of work to deny creation requests against a terminating namespace while avoiding flakes

### CODE REVIEW

File: pkg/project/admission/lifecycle/admission.go [hunk 1]
```diff
@@ -47,12 +48,15 @@ var recommendedCreatableResources = map[unversioned.GroupResource]bool{
...
 
-// Admit enforces that a namespace must exist in order to associate content with it.
-// Admit enforces that a namespace that is terminating cannot accept new content being associated with it.
+// Admit enforces that a namespace must have the openshift finalizer associated with it in order to create origin API objects within it
 func (e *lifecycle) Admit(a admission.Attributes) (err error) {
 	if len(a.GetNamespace()) == 0 {
 		return nil
 	}
+	// only pay attention to origin resources
+	if latest.OriginKind(a.GetKind()) {
```
Comment (derekwaynecarr) [inline pkg/project/admission/lifecycle/admission.go]: flip this

**Comment by derekwaynecarr** [comment]
[merge]

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
networking failed with unrelated error (`Error: Failed to synchronize cache for repo 'fedora'`)
green tests otherwise
[merge]

**Comment by liggitt** [comment]
LimitRange test was at 160 seconds

**Comment by kargakis** [comment]
Should this also fix https://github.com/openshift/origin/issues/11544?

**Pull request merged**